### PR TITLE
[Fluid] Unify serial mpi defaults

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**Description**
This finally unifies the serial / MPI json settings in the `FluidDynamicsApplication`.

### CODE REVIEW

**Comment by philbucher** [comment]
awesome!
I will take a look asap!

File: applications/FluidDynamicsApplication/python_scripts/python_solvers_wrapper_fluid.py [hunk 1]
```diff
@@ -35,24 +35,24 @@ def CreateSolverByParameters(model, solver_settings, parallelism):
...
+        elif solver_type == "two_fluids" or solver_type == "TwoFluids":
             solver_module_name = "navier_stokes_two_fluids_solver"
 
         else:
             raise Exception("the requested solver type is not in the python solvers wrapper. Solver type is : " + solver_type)
 
     # Solvers for MPI parallelism
     elif (parallelism == "MPI"):
-        if (solver_type == "Monolithic"):
+        if solver_type == "monolithic" or solver_type == "Monolithic":
```
Comment (philbucher) [inline applications/FluidDynamicsApplication/python_scripts/python_solvers_wrapper_fluid.py:46]: :heart: :heart: :heart: :heart: :heart:

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py [hunk 2]
```diff
@@ -303,7 +303,6 @@ def ValidateSettings(self):
             self.settings["fm_ale_settings"]["fm_ale_solver_settings"].ValidateAndAssignDefaults(self._get_fm_ale_solver_default_settings(mesh_movement))
 
     def __init__(self, model, custom_settings):
-        self._validate_settings_in_baseclass=True # To be removed eventually
```
Comment (sunethwarna) [inline applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py:306]: Are we removing this now? Finally :) :+1:

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py:306]: Actually, I think it hasn't been used for a while. :smile:

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py [hunk 3]
```diff
@@ -303,7 +303,6 @@ def ValidateSettings(self):
...
         # TODO: DO SOMETHING IN HERE TO REMOVE THE "time_order" FROM THE DEFAULT SETTINGS BUT KEEPING THE BACKWARDS COMPATIBILITY
         super(NavierStokesEmbeddedMonolithicSolver,self).__init__(model,custom_settings)
```
Comment (sunethwarna) [inline applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py:307]:
```suggestion
        super().__init__(model,custom_settings)
```

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_solver_fractionalstep.py [hunk 4]
```diff
@@ -89,7 +89,6 @@ def GetDefaultParameters(cls):
         return default_settings
 
     def __init__(self, model, custom_settings):
-        self._validate_settings_in_baseclass=True # To be removed eventually
         super(NavierStokesSolverFractionalStep,self).__init__(model,custom_settings)
```
Comment (sunethwarna) [inline applications/FluidDynamicsApplication/python_scripts/navier_stokes_solver_fractionalstep.py:92]:
```suggestion
        super().__init__(model,custom_settings)
```

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_solver_vmsmonolithic.py [hunk 5]
```diff
@@ -271,7 +271,6 @@ def _BackwardsCompatibilityHelper(self,settings):
 
 
     def __init__(self, model, custom_settings):
-        self._validate_settings_in_baseclass=True # To be removed eventually
         custom_settings = self._BackwardsCompatibilityHelper(custom_settings)
         super(NavierStokesSolverMonolithic,self).__init__(model,custom_settings)
```
Comment (sunethwarna) [inline applications/FluidDynamicsApplication/python_scripts/navier_stokes_solver_vmsmonolithic.py:275]:
```suggestion
        super().__init__(model,custom_settings)
```

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py [hunk 6]
```diff
@@ -80,10 +80,7 @@ def GetDefaultParameters(cls):
         return default_settings
 
     def __init__(self, model, custom_settings):
-        self._validate_settings_in_baseclass=True # To be removed eventually
-
         # TODO: DO SOMETHING IN HERE TO REMOVE THE "time_order" FROM THE DEFAULT SETTINGS BUT KEEPING THE BACKWARDS COMPATIBILITY
-
         super(NavierStokesTwoFluidsSolver,self).__init__(model,custom_settings)
```
Comment (sunethwarna) [inline applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py:84]:
```suggestion
        super().__init__(model,custom_settings)
```

File: applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_solver_fractionalstep.py [hunk 7]
```diff
@@ -16,73 +16,12 @@ def CreateSolver(model, custom_settings):
...
-        self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DYNAMIC_TAU, self.settings["dynamic_tau"].GetDouble())
+        # Call the serial base class constructor
+        super().__init__(model,custom_settings)
 
         KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__,"Construction of TrilinosNavierStokesSolverFractionalStep solver finished.")
 
-
     def AddVariables(self):
         ## Add variables from the base class
         super(TrilinosNavierStokesSolverFractionalStep, self).AddVariables()
```
Comment (sunethwarna) [inline applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_solver_fractionalstep.py:27]:
```suggestion
        super().AddVariables()
```

**Review by sunethwarna — COMMENTED** [review]
Minor missed stuff. I'll leave the approval to @philbucher :)

**Comment by philbucher** [comment]
@sunethwarna : #7451

**Comment by rubenzorrilla** [comment]
> Minor missed stuff. I'll leave the approval to @philbucher :)

Well, I intentionally left all these since where lines not required to be modified for the purpose of this PR. I'd do all these changes at once in a single PR.

**Review by philbucher — APPROVED** [review]
:+

**Pull request merged**